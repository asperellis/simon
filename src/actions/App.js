// types
export const GET_USER_LOCATION = 'GET_USER_LOCATION';

// creators
export function getUserLocation() {
  const defaultAction = {
    type: GET_USER_LOCATION,
    location: null
  };

  return dispatch => {
    if (navigator) {
      const geolocation = navigator.geolocation;
      geolocation.getCurrentPosition(
        position => {
          dispatch({
            ...defaultAction,
            location: position.coords
          });
        },
        () => {
          dispatch(defaultAction);
        },
        {
          enableHighAccuracy: true,
          timeout: 10000,
          maximumAge: 0
        }
      );
    } else {
      dispatch(defaultAction);
    }
  };
}
