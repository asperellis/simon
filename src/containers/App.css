.appContent {
  margin-top: 74px;
  min-height: 100vh;
}
