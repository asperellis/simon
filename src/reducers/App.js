import { combineReducers } from 'redux';

const location = (state = null, action) => {
  if (action.type === 'GET_USER_LOCATION') {
    return action.location;
  }

  return state;
};

const appReducer = combineReducers({
  location
});

export default appReducer;
