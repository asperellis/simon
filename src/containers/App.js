import React, { Component } from 'react';
import styles from './App.css';
import Header from './../components/Header/Header';
import Footer from './../components/Footer/Footer';
import { connect } from 'react-redux';
import { getUserLocation } from './../actions/App';

const mapStateToProps = state => {
  return {
    location: state.location
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getUserLocation: () => dispatch(getUserLocation())
  };
};

class AppShell extends Component {
  render() {
    return (
      <div className={styles.app}>
        <Header
          search={true}
          location={this.props.location}
          getUserLocation={this.props.getUserLocation}
        />
        <main className={styles.appContent}>
          <div className="container">
            <div className="row">
              <div className="col-md-12">
                <h1>This is a React Simon Wrapper</h1>
                <p>
                  {this.props.location &&
                    'Your Location is Lat: ' +
                      (this.props.location && this.props.location.latitude) +
                      ' Long: ' +
                      (this.props.location && this.props.location.longitude)}
                </p>
                <p>To test out the possibility of using on business site.</p>
                <p>Data and Store handling still needed</p>
              </div>
            </div>
          </div>
        </main>
        <Footer />
      </div>
    );
  }
}

const App = connect(mapStateToProps, mapDispatchToProps)(AppShell);

export default App;
