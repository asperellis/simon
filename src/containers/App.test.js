import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

test('Text Changes', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  component.root.findByType('button').props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
