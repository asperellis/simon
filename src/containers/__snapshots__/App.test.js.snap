// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Changes 1`] = `
<div>
  <h1>
    Hey!
  </h1>
  <button
    onClick={[Function]}
  >
    Change Text
  </button>
</div>
`;

exports[`Text Changes 2`] = `
<div>
  <h1>
    Yo!
  </h1>
  <button
    onClick={[Function]}
  >
    Change Text
  </button>
</div>
`;
